pkg/hyper/helper.go

```diff
@@ -271,3 +292,53 @@ func toKubeContainerState(state string) kubeapi.ContainerState {
 		return kubeapi.ContainerState_CONTAINER_UNKNOWN
 	}
 }
+
+// TODO(harry) These two methods will find subsystem mount point frequently, consider move FindCgroupMountpoint into a unified place.
+// getCpuLimitFromCgroup get the cpu limit from given cgroupParent
+func getCpuLimitFromCgroup(cgroupParent string) (int32, error) {
+	mntPath, err := libcontainercgroups.FindCgroupMountpoint("cpu")
+	if err != nil {
+		return -1, err
+	}
+	contents, err := ioutil.ReadFile(filepath.Join(mntPath, cgroupParent, cpuQuotaCgroupFile))
+	if err != nil {
+		return -1, err
+	}
+
+	cpuQuotaStr := strings.TrimSpace(string(contents))
+
+	if value, err := strconv.Atoi(cpuQuotaStr); err == nil {
+		// k8s use 100 milicpu to describe cpu quota
+		vcpuNumber := value / cpu100m
+		// but hyper only support int32 vcpu number, should be one vcpu at least
+		if vcpuNumber < 1 {
+			vcpuNumber = 1
+		}
+		return int32(vcpuNumber), nil
+	} else {
+		return -1, err
+	}
+}
+
+// getMemeoryLimitFromCgroup get the memory limit from given cgroupParent
+func getMemeoryLimitFromCgroup(cgroupParent string) (int32, error) {
+	mntPath, err := libcontainercgroups.FindCgroupMountpoint("memory")
+	if err != nil {
+		return -1, err
+	}
+	// TODO(harry) k8s does not forbidden illegal number e.g. `512m`, this will create a file with only `4096` bytes Memory
+	// need to fix this on upstream.
+	contents, err := ioutil.ReadFile(filepath.Join(mntPath, cgroupParent, memoryCgroupFile))
+	if err != nil {
+		return -1, err
+	}
+
+	memoryinMegabytesStr := strings.TrimSpace(string(contents))
+
+	if value, err := strconv.Atoi(memoryinMegabytesStr); err == nil {
+		memoryinMegabytes := int32(value / MiB)
```

Comment:
Hyper needs a minimum value of memory to keep sandbox and container up, e.g. 64MB)